pkg/system/phase4_configuring.go

```diff
@@ -431,25 +432,52 @@ func (r *Reconciler) setDesiredEndpointMounts(podSpec *corev1.PodSpec, container
 			isMountExist := false
 			for _, volumeMount := range container.VolumeMounts {
 				if volumeMount.Name == volumeName && volumeMount.SubPath == subPath {
-						isMountExist = true // volumeMount already created - no need to add
-						break;
+					isMountExist = true // volumeMount already created - no need to add
+					break
 				}
 			}
-			if (!isMountExist) {
+			if !isMountExist {
 				container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{
-					Name: volumeName,
+					Name:      volumeName,
 					MountPath: mountPath,
-					SubPath: subPath,
+					SubPath:   subPath,
 				})
 			}
 		}
 	}
 	return nil
 }
 
-// ReconcileHPAEndpoint reconcile the endpoint's HPS and report the configuration
+// awaitEndpointDeploymentPods wait for the the endpoint deployment to become ready
+// before creating the controlling HPA
+// See https://bugzilla.redhat.com/show_bug.cgi?id=1885524
+func (r *Reconciler) awaitEndpointDeploymentPods() error {
+
+	// Check that all deployment pods are available
+	availablePods := r.DeploymentEndpoint.Status.AvailableReplicas
+	desiredPods := r.DeploymentEndpoint.Status.Replicas
+	if availablePods == 0 || availablePods != desiredPods {
+		return errors.New("not enough available replicas in endpoint deployment")
+	}
+
+	// Check that deployment is ready
+	for _, condition := range r.DeploymentEndpoint.Status.Conditions {
+		if condition.Status != "True" {
+			return errors.New("endpoint deployment is not ready")
+		}
+	}
+
+	return nil
+}
+
+// ReconcileHPAEndpoint reconcile the endpoint's HPA and report the configuration
 // back to the noobaa core
 func (r *Reconciler) ReconcileHPAEndpoint() error {
+	// Wait for the the endpoint deployment to become ready
+	if err := r.awaitEndpointDeploymentPods(); err != nil {
```

Comment:
I think that always waiting for endpoints deployment before reconciling HPA, can cause us to get stuck here if the deployment can't reach the desired number of replicas (e.g. if the user sets a high number of min count in the spec).

Reply:
@dannyzaken  good point, condition to wait only if HPA was not created yet was added.